lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -409,175 +508,186 @@ internal void WriteContentTo(XmlWriter writer, string elementName, SyndicationCo
         {
             for (int i = 0; i < contributors.Count; ++i)
             {
                 SyndicationPerson p = contributors[i];
-                WritePersonTo(writer, p, Atom10Constants.ContributorTag);
+                await WritePersonToAsync(XmlWriterWrapper.CreateFromWriter(writer), p, Atom10Constants.ContributorTag);
             }
         }
 
-        internal void WriteFeedLastUpdatedTimeTo(XmlWriter writer, DateTimeOffset lastUpdatedTime, bool isRequired)
+        internal async Task WriteFeedLastUpdatedTimeToAsync(XmlWriterWrapper writer, DateTimeOffset lastUpdatedTime, bool isRequired)
         {
             if (lastUpdatedTime == DateTimeOffset.MinValue && isRequired)
             {
                 lastUpdatedTime = DateTimeOffset.UtcNow;
             }
             if (lastUpdatedTime != DateTimeOffset.MinValue)
             {
-                WriteElement(writer, Atom10Constants.UpdatedTag, AsString(lastUpdatedTime));
+                await WriteElementAsync(writer, Atom10Constants.UpdatedTag, AsString(lastUpdatedTime));
             }
         }
 
-        internal void WriteItemAuthorsTo(XmlWriter writer, Collection<SyndicationPerson> authors)
+        internal async Task WriteItemAuthorsToAsync(XmlWriter writer, Collection<SyndicationPerson> authors)
         {
+            XmlWriterWrapper wrappedWriter = XmlWriterWrapper.CreateFromWriter(writer);
             for (int i = 0; i < authors.Count; ++i)
             {
                 SyndicationPerson p = authors[i];
-                WritePersonTo(writer, p, Atom10Constants.AuthorTag);
+                await WritePersonToAsync(wrappedWriter, p, Atom10Constants.AuthorTag);
             }
         }
 
-        internal void WriteItemContents(XmlWriter dictWriter, SyndicationItem item)
+        internal async Task WriteItemContentsAsync(XmlWriter dictWriter, SyndicationItem item)
         {
-            WriteItemContents(dictWriter, item, null);
+            await WriteItemContentsAsync(XmlWriterWrapper.CreateFromWriter(dictWriter), item, null);
         }
 
-        internal void WriteItemContributorsTo(XmlWriter writer, Collection<SyndicationPerson> contributors)
+        internal async Task WriteItemContributorsToAsync(XmlWriter writer, Collection<SyndicationPerson> contributors)
         {
+            XmlWriterWrapper wrappedWriter = XmlWriterWrapper.CreateFromWriter(writer);
+
             for (int i = 0; i < contributors.Count; ++i)
             {
                 SyndicationPerson p = contributors[i];
-                WritePersonTo(writer, p, Atom10Constants.ContributorTag);
+                await WritePersonToAsync(wrappedWriter, p, Atom10Constants.ContributorTag);
             }
         }
 
-        internal void WriteItemLastUpdatedTimeTo(XmlWriter writer, DateTimeOffset lastUpdatedTime)
+        internal async Task WriteItemLastUpdatedTimeToAsync(XmlWriterWrapper writer, DateTimeOffset lastUpdatedTime)
         {
```

Comment:
Blank line after if block. Remove async and await, return Task from WriteElementStringAsync